### PR TITLE
Refactoring edit_content out

### PR DESCRIPTION
**Pull request by lygstate**

### CODE REVIEW

**Comment by lygstate** [comment]
Can this be merged first? then base on this to update  #100

File: git-publish [hunk 1]
```diff
@@ -446,13 +459,11 @@ def menu_select(menu):
...
-    tmpfile.write(os.linesep.join(cc_list).encode(ENCODING))
-    tmpfile.flush()
-    edit(tmpfile.name)
+    new_content_bytes = edit_content(os.linesep.join(cc_list).encode(ENCODING), '.txt')
+    new_content = new_content_bytes.decode(ENCODING)
     r = []
-    for line in open(tmpfile.name, "r").readlines():
-        r += [x.strip() for x in line.split(",")]
+    for line in new_content.splitlines():
+        r += [x.strip() for x in line.split(",") if len(x.strip())]
```
Comment (stefanha) [inline git-publish:466]: Why was `if len(x.strip())` added? I'm not sure if this fixes a pre-existing issue or something related to switching to edit_content().

File: git-publish [hunk 2]
```diff
@@ -420,6 +420,19 @@ def edit(*filenames):
...
+    fd, tmpfile = tempfile.mkstemp(suffix=suffix)
+    fd_opened = os.fdopen(fd, 'wb')
+    fd_opened.write(content)
+    fd_opened.close()
+    edit(tmpfile)
+    fd_opened = open(tmpfile, "rb")
+    new_content = fd_opened.read()
+    fd_opened.close()
+    if tmpfile:
+        os.unlink(tmpfile)
```
Comment (stefanha) [inline git-publish:433]: Please use `try: ... finally: ...` to ensure that the temporary file is unlinked even if an exception occurs. See `tag()` for an example.

File: git-publish (hunk 1 above)
Comment (lygstate) [reply git-publish:470]: # Remove blank email item in list by len(x.strip())

Comment (stefanha) [reply git-publish:471]: Why would there be ",," blank items in the comma-separated list? Is there a reason why it was necessary to add this?

Comment (lygstate) [reply git-publish:471]: > Why would there be ",," blank items in the comma-separated list? Is there a reason why it was necessary to add this?

It's just a normalize, such as mis-typing, it's doesn't hurt anything

Comment (stefanha) [reply git-publish:471]: I see. Please put unrelated changes into separate commits. That way it's easier to review the code and understand the reason for the changes.

Comment (lygstate) [reply git-publish:471]: OK, split into two commits

Comment (stefanha) [reply git-publish:471]: Thanks for your patience!

**Pull request merged**